src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -16,11 +16,13 @@ internal class CosmosDBTriggerAttributeBindingProvider : ITriggerBindingProvider
         private readonly string _monitorConnectionString;
         private readonly string _leasesConnectionString;
         private readonly ChangeFeedHostOptions _leasesOptions;
+        private readonly INameResolver _nameResolver;
 
-        public CosmosDBTriggerAttributeBindingProvider(string monitorConnectionString, string leasesConnectionString, ChangeFeedHostOptions leasesOptions = null)
+        public CosmosDBTriggerAttributeBindingProvider(INameResolver nameResolver, ChangeFeedHostOptions leasesOptions = null)
         {
-            _monitorConnectionString = monitorConnectionString;
-            _leasesConnectionString = leasesConnectionString;
+            _nameResolver = nameResolver;
+            _monitorConnectionString = _nameResolver.Resolve(DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName);
+            _leasesConnectionString = _nameResolver.Resolve(DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName);
```

Comment:
Can you move these two to `TryCreateAsync`? I don't think it's guaranteed that this call won't throw, and we wouldn't want that to happen in a constructor.